src/BloomTests/WebLibraryIntegration/BloomS3ClientTests.cs

```diff
@@ -67,6 +72,27 @@ public void UploadBook_Simple_FilesAreOnS3InExpectedDirectory()
 			}
 		}
 
+		[Test]
+		public void UploadBook_ContainsThumbsFile_DontCopyItToS3()
+		{
+			string srcBookPath = MakeBook();
+			AddThumbsFile(srcBookPath);
+			const string excludedFile = "thumbs.db";
+			var storageKeyOfBookFolder = UploadBook(srcBookPath);
+
+			// It's possible that another unit test uploads a book at the same time, so we can't
+			// test for strict equality.
+			Assert.That(_client.GetCountOfAllFilesInBucket(), Is.AtLeast(Directory.GetFiles(srcBookPath).Count()));
+
+			var expectedDir = Path.GetFileName(srcBookPath);
+			foreach (var file in Directory.GetFiles(srcBookPath))
```

Comment:
Interesting... my first inclination was actually to just add the thumbs.db check to the existing test, but I had the same "concerns" qualm. But in light of the time factor I:
1- amalgamated my test into the existing UploadBook_Simple_FilesAreOnS3InExpectedDirectory()
2- made a Debt card with your "Another option..." paragraph